### ISSUE TITLE
Registering Async Reply Enabled Async Human Input

### ISSUE THREAD

**Issue by ShobhitVishnoi30** [microsoft/autogen#806]
The main issue that would occur without registering the async reply function is:

Async Human Input Would Not Work

Without registering ConversableAgent.a_check_termination_and_human_reply as a possible reply, the generate_reply method would never call the async reply functions like a_get_human_input. So any async human input flows would be broken - the agent would not be able to get async human input properly.

So in summary, the key problem was:

- Async human input methods like a_get_human_input would not be called
- Therefore no async human input would work
- User would not be able to provide async inputs to the agent

Registering that additional async reply handler fixed this issue and enabled the async human input flow.
